File: src/main/java/org/foo/dto/ProjectDto.java
package org.foo.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.foo.enums.Status;

import java.time.LocalDate;

@NoArgsConstructor
@AllArgsConstructor
@Data
public class ProjectDto {

  private Long id;

  private String projectName;

  private String projectCode;

  private UserDto assignedManager;

  private LocalDate startDate;

  private LocalDate endDate;

  private String projectDetail;

  private Status projectStatus;

  private int completeTaskCounts;
  private int unfinishedTaskCounts;



}

File: lab1-frontend/src/app/common/project-dto.ts
import {User} from "./user";
import {Status} from "../enums/status";

export class ProjectDto {
  constructor(
    public id: number,
    public projectName: string,
    public projectCode: string,
    public assignedManager: User,      // Reference to UserDto
    public startDate: Date,                // Using Date type for start date
    public endDate: Date,                  // Using Date type for end date
    public projectDetail: string,
    public projectStatus: Status,          // Assuming Status is an enum
    public completeTaskCounts: number,
    public unfinishedTaskCounts: number
  ) {}
}
